.github/pull_request_template.md

```diff
@@ -1,35 +1,83 @@
+
+Before submitting your PR, please ensure:
+- It's linked to an approved GitHub Issue.
+- You've reviewed our [Contributing Guidelines](https://docs.roocode.com/contributing).
+-->
+
+### Related GitHub Issue
 
-<!-- Brief description of WHAT you’re doing and WHY. -->
+<!-- Every PR MUST be linked to an approved issue. -->
 
-## Implementation
+Closes: # <!-- Replace with the issue number, e.g., Closes: #123 -->
+
+### Description
 
 <!--
+Briefly summarize the changes in this PR and how they address the linked issue.
+The issue should cover the "what" and "why"; this section should focus on:
+- The "how": key implementation details, design choices, or trade-offs made.
+- Anything specific reviewers should pay attention to in this PR.
+-->
 
-Some description of HOW you achieved it. Perhaps give a high level description of the program flow. Did you need to refactor something? What tradeoffs did you take? Are there things in here which you’d particularly like people to pay close attention to?
+### Test Procedure
 
+<!--
+Detail the steps to test your changes. This helps reviewers verify your work.
+- How did you test this specific implementation? (e.g., unit tests, manual testing steps)
+- How can reviewers reproduce your tests or verify the fix/feature?
+- Include relevant testing environment details if applicable.
 -->
 
-## Screenshots
+### Type of Change
 
-| before | after |
-| ------ | ----- |
-|        |       |
+<!-- Mark all applicable boxes with an 'x'. -->
 
-## How to Test
+- [ ] 🐛 **Bug Fix**: Non-breaking change that fixes an issue.
+- [ ] ✨ **New Feature**: Non-breaking change that adds functionality.
+- [ ] 💥 **Breaking Change**: Fix or feature that would cause existing functionality to not work as expected.
+- [ ] ♻️ **Refactor**: Code change that neither fixes a bug nor adds a feature.
+- [ ] 💅 **Style**: Changes that do not affect the meaning of the code (white-space, formatting, etc.).
+- [ ] 📚 **Documentation**: Updates to documentation files.
+- [ ] ⚙️ **Build/CI**: Changes to the build process or CI configuration.
+- [ ] 🧹 **Chore**: Other changes that don't modify `src` or test files.
 
-<!--
+### Pre-Submission Checklist
+
+<!-- Go through this checklist before marking your PR as ready for review. -->
+
+- [ ] **Issue Linked**: This PR is linked to an approved GitHub Issue (see "Related GitHub Issue" above).
+- [ ] **Scope**: My changes are focused on the linked issue (one major feature/fix per PR).
+- [ ] **Self-Review**: I have performed a thorough self-review of my code.
+- [ ] **Code Quality**:
+    - [ ] My code adheres to the project's style guidelines.
+    - [ ] There are no new linting errors or warnings (`npm run lint`).
+    - [ ] All debug code (e.g., `console.log`) has been removed.
+- [ ] **Testing**:
+    - [ ] New and/or updated tests have been added to cover my changes.
+    - [ ] All tests pass locally (`npm test`).
+    - [ ] The application builds successfully with my changes.
```

Comment:
Not a blocker, but I'd love to turn as many of these into automated checks as we can